app/src/main/java/io/intrepid/contest/screens/contestcreation/describecontest/DescribeContestFragment.java

```diff
@@ -27,9 +31,35 @@ public DescribeContestPresenter createPresenter(PresenterConfiguration configura
         return new DescribeContestPresenter(this, configuration);
     }
 
+    @OnTextChanged(R.id.contest_description_edittext)
+    public final void onTextChanged(CharSequence newDesription) {
+        if (TextUtils.isEmpty(newDesription)) {
+            onNextInvalidated();
+        } else {
+            onNextValidated();
+        }
+    }
+
+    @Override
+    public void enableNext() {
+        EditContestContract activity = (EditContestContract) getActivity();
+        activity.onNextEnabled();
+    }
+
+    @Override
+    public void disableNext() {
+        EditContestContract activity = (EditContestContract) getActivity();
+        activity.onNextDisabled();
+    }
+
+    @Override
+    public void onNextInvalidated() {
+        presenter.onNextInvalidated();
```

Comment:
we can remove this presenter function if we refactor the chunk of code Gabi mentioned.